src/Files.App/Shell/Win32API.cs

```diff
@@ -16,13 +19,75 @@
 using System.Threading.Tasks;
 using System.Windows.Forms;
 using Vanara.PInvoke;
+using Windows.ApplicationModel.AppService;
+using Windows.Foundation.Collections;
 using Windows.System;
 
 namespace Files.App.Shell
 {
     [SupportedOSPlatform("Windows10.0.10240")]
     internal class Win32API
     {
+        public static Task StartSTATask(Func<Task> func)
+        {
+            var tcs = new TaskCompletionSource();
+            Thread thread = new Thread(async () =>
+            {
+                Ole32.OleInitialize();
+                try
+                {
+                    await func();
+                    tcs.SetResult();
+                }
+                catch (Exception ex)
+                {
+                    tcs.SetResult();
+                    App.Logger.Warn(ex, ex.Message);
+                }
+                finally
+                {
+                    Ole32.OleUninitialize();
+                }
+            })
+            {
+                IsBackground = true,
+                Priority = ThreadPriority.Normal
+            };
+            thread.SetApartmentState(ApartmentState.STA);
+            thread.Start();
+            return tcs.Task;
+        }
+
+        public static Task StartSTATask(Action action)
+        {
+            var tcs = new TaskCompletionSource();
```

Comment:
```suggestion
            var taskCompletionSource = new TaskCompletionSource();
```
Can you do the same for the other variables? It helps make the code more readable.